src/transformers/integrations/tensor_parallel.py

```diff
@@ -1014,7 +1014,8 @@ def shard_and_distribute_module(
 
     """
     param_name, param_type = parameter_name.rsplit(".", 1) if "." in parameter_name else parameter_name
-    tp_plan = model._tp_plan
+    tp_plan = model._tp_plan or {}
+    tp_plan.update(getattr(type(model), "_tp_plan", {}))
```

Comment:
this happens after the first call to `distribute_module` meaning the plan for lm head won't make it (won't have hooks)